contracts/orderValidation/OrderValidatorV1.sol

```diff
@@ -0,0 +1,439 @@
+     * @param collection address of the collection
+     * @param user address of the user
+     * @param tokenId tokenId
+     * @param amount amount
+     */
+    function _validateNFTApprovals(
+        address collection,
+        address user,
+        uint256 tokenId,
+        uint256 amount
+    ) internal view returns (uint256 validationCode) {
+        address transferManager;
+
+        if (IERC165(collection).supportsInterface(INTERFACE_ID_ERC721)) {
+            transferManager = TRANSFER_MANAGER_ERC721;
+        } else if (IERC165(collection).supportsInterface(INTERFACE_ID_ERC1155)) {
+            transferManager = TRANSFER_MANAGER_ERC1155;
+        } else {
+            transferManager = transferSelectorNFT.transferManagerSelectorForCollection(collection);
+        }
+
+        if (transferManager == address(0)) return NO_TRANSFER_MANAGER_AVAILABLE_FOR_COLLECTION;
+
+        if (transferManager == TRANSFER_MANAGER_ERC721) {
+            _validateERC721AndEquivalents(collection, user, transferManager, tokenId);
+        } else if (transferManager == TRANSFER_MANAGER_ERC1155) {
+            _validateERC1155(collection, user, transferManager, tokenId, amount);
+        } else {
+            return NO_TRANSFER_MANAGER_AVAILABLE_FOR_COLLECTION;
+        }
+    }
+
+    /**
+     * @notice Check validity of ERC20 approvals and balances required to process the order
+     * @param currency address of the currency
+     * @param user address of the user
+     * @param price price
+     */
+    function _validateERC20(
+        address currency,
+        address user,
+        uint256 price
+    ) internal view returns (uint256 validationCode) {
+        if ((IERC20(currency).allowance(user, address(looksRareExchange))) < price)
+            return ERC20_APPROVAL_INFERIOR_TO_AMOUNT;
+        if ((IERC20(currency).balanceOf(user)) < price) return ERC20_BALANCE_INFERIOR_TO_AMOUNT;
+    }
+
+    /**
+     * @notice Check validity of ERC721 approvals and balances required to process the order
+     * @param collection address of the collection
+     * @param user address of the user
+     * @param transferManager address of the transfer manager
+     * @param tokenId tokenId
+     */
+    function _validateERC721AndEquivalents(
+        address collection,
+        address user,
+        address transferManager,
+        uint256 tokenId
+    ) internal view returns (uint256 validationCode) {
+        (bool answer, bytes memory data) = collection.staticcall(
+            abi.encodeWithSelector(IERC721.getApproved.selector, tokenId)
+        );
+
+        address approvedAddress;
+
+        if (answer) {
+            approvedAddress = abi.decode(data, (address));
+        }
+
+        bool isApprovedSingle = approvedAddress == transferManager;
+
+        (answer, data) = collection.staticcall(
+            abi.encodeWithSelector(IERC721.isApprovedForAll.selector, user, transferManager)
+        );
+
+        bool isApprovedAll;
```

Comment:
I understand it is a view only contract, but just in case if in the future someone decides to inherit this contract for other purposes? (and we might potentially reuse some logic in a V2 validator?)

Reply:
Good catch, it is adjusted!